### PR TITLE
Add common notes feature

### PR DESCRIPTION
**Pull request by NShani**
**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [X] Your code is flake8 compliant 
- [X] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [X] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR. 
[Feature Documentation - Common Note Modification.pdf](https://github.com/DefectDojo/django-DefectDojo/files/4349211/DefectDojo.-.Common.Note.Modification1.pdf)

- [X] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [X] Add the proper label to categorize your PR 
In the current DefectDojo application, there is a functionality for the users to add notes to findings. In this modification, the capability of adding a note for similar findings is introduced. 

This feature is useful when similar common findings are reported in the new scan reports, to automatically add the notes to the new finding associated with those old specific findings.

As an example, whenever a Veracode scan is performed, the ‘Improper Neutralization of CRLF Sequences ('CRLF Injection')’ issue is reported against all the products. But this issue has been already mitigated globally via the tomcat level. Therefore when this issue is reported, it should be associated with a mitigation note explaining that this issue is already addressed and it does not have any impact on the product. Manually adding the same note again and again for the findings of new scan reports, is not scalable. 
With this feature, system admin can identify specific findings and add a common note to associate with it. Then that note will appear in all the similar findings reported in other scan reports. 

The common note modification has no effect on new or existing installations, or the existing
functionalities of the DefectDojo application. ([DefectDojo - Common Note Modification.pdf](https://github.com/DefectDojo/django-DefectDojo/files/4349211/DefectDojo.-.Common.Note.Modification1.pdf))

Current accepted labels for PRs:
- feature

### CODE REVIEW

**Comment by madchap** [comment]
Didn't really look, but is this related to https://github.com/DefectDojo/django-DefectDojo/issues/2069?

**Comment by valentijnscholten** [comment]
Could you describe why this is needed next to the Risk Acceptance feature? The Risk Acceptance features allows to add a risk acceptance (with note(s) to a set of findings. And will have a bulk option soon, at least via API in #1904

**Comment by NShani** [comment]
> Could you describe why this is needed next to the Risk Acceptance feature? The Risk Acceptance features allows to add a risk acceptance (with note(s) to a set of findings. And will have a bulk option soon, at least via API in #1904

What this feature does is allowing automatically add the 'Notes' for specific findings when we import a scan report.

As an example, think of a scenario where a static scanner reports ‘Improper Neutralization of
CRLF Sequences in HTTP Headers ('HTTP Response Splitting')’ issue multiple times across a
code base, when there is a mitigation already available globally via a common security later (for
example: Apache Tomcat automatically prevents CRLF attacks, by applying required filtering of
HTTP headers). In this scenario, when the issue is reported, it should be associated with a
mitigation note explaining that this issue is already addressed and it does not have any impact
on the security of the product. Manually adding the same note again and again for the findings
of new scan reports is not scalable.

With this new feature, system admin can create a common note to be associated with a type of
a finding. The defined note will be automatically added to all the similar findings reported in
future scan reports. Common notes can be defined based on the "issue title", "CWE" and the
"scanner".

In the documentation this has been explained with an example scenario.

**Comment by madchap** [comment]
I quickly tried to pull your PR to give it a whirl anyways, got this after performing the migration (which does need a rebase):

`(1146, "Table 'defectdojo.dojo_commonnote' doesn't exist")`

Looking at the database, it looks like the table is named differently.

```
MySQL [defectdojo]> show tables like '%common%';
+---------------------------------+
| Tables_in_defectdojo (%common%) |
+---------------------------------+
| dojo_common_note                |
+---------------------------------+
```

Can you please check (and rebase)? Thank you.

**Comment by NShani** [comment]
@madchap updated the migration script for the modified table. Could you please check now? Thank you.

**Comment by NShani** [comment]
@madchap There was a conflict and I resolved it. Could you please check this PR now? 
Thank you.

**Comment by Maffooch** [comment]
Hi @NShani please look into the failing travis tests for further changes. Moderators will revisit once all three tests are passings.

**Comment by NShani** [comment]
@Maffooch Fixed the travis failing. Could you please check this PR now?
Thank you.

**Comment by Maffooch** [comment]
This is a cool feature! The only thing Im not sold on is making all fields required in the initialization. CWE especially since it means scanners/parser that do not provide CWE can't be used with common notes. Also I got this error (UnboundLocalError at /common/note/1/view1
local variable 'available_note_types' referenced before assignment)

**Comment by NShani** [comment]
@Maffooch I tried to reproduce the error you mentioned. But I couldn't. Could you please let me know the steps to reproduce the issue. I will check. 
Thank You.

**Comment by NShani** [comment]
@Maffooch Highly appreciates if you could let me know the steps to reproduce the issue.

Thank You.

**Comment by damiencarol** [comment]
@NShani made a quick review and I have 2 comments:
1) could you rebase your branch?
2) You add another big lits of all the parsers, could you reuse an existing one ?

File: dojo/forms.py [hunk 1]
```diff
@@ -2067,3 +2067,95 @@ class LoginBanner(forms.Form):
...
+                         ("Wpscan", "Wpscan"),
+                         ("Sslscan", "Sslscan"),
+                         ("JFrog Xray Scan", "JFrog Xray Scan"),
+                         ("Sslyze Scan", "Sslyze Scan"),
+                         ("Testssl Scan", "Testssl Scan"),
+                         ("Hadolint Dockerfile check", "Hadolint Dockerfile check"),
+                         ("Aqua Scan", "Aqua Scan"),
+                         ("HackerOne Cases", "HackerOne Cases"),
+                         ("Xanitizer Scan", "Xanitizer Scan"),
+                         ("Trivy Scan", "Trivy Scan"))
```
Comment (damiencarol) [inline dojo/forms.py:2145]: Your pull request add another big list of parsers to maintain. we should try to do it differently

File: dojo/common_note/urls.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+urlpatterns = [
+    url(r'^common_note',
+        views.common_note, name='common_note'),
+    url(r'^add_common_note',
+        views.add_common_note, name='add_common_note'),
+    url(r'^common/note/(?P<cnid>\d+)/note1$',
+        views.add_common_note_comment, name='add_common_note_comment'),
+    url(r'^common/note/(?P<cnid>\d+)/edit$',
+        views.edit_common_note, name='edit_common_note'),
+    url(r'^common/note/(?P<cnid>\d+)/view1$',
```
Comment (damiencarol) [inline dojo/common_note/urls.py:14]: could you use `common_note/foo/bar` to be aligned with object name?

**Review by damiencarol — CHANGES_REQUESTED** [review]

**Pull request closed without merging**